src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -1086,7 +1086,11 @@ public function relationLoaded($key)
         }
 
         if ($nestedRelation !== null) {
-            foreach ($this->$relation as $related) {
+            $relatedModels = is_iterable($relatedModels = $this->$relation)
+                ? $relatedModels
+                : [$relatedModels];
+
+            foreach ($relatedModels as $related) {
                 if (! $related->relationLoaded($nestedRelation)) {
```

Comment:
@rodrigopedra if one of the relations is not loaded, or does not exist, `$relatedModels` can be `[null]` leading to `Error: Call to a member function relationLoaded() on null`

Reply:
You are right, thanks for the heads-up, I will send a new PR addressing this case.

Reply:
@nicekiwi sent PR #55531 thanks again!